static/app/utils/makeSafeRefluxStore.ts

```diff
@@ -0,0 +1,50 @@
+export interface SafeRefluxStore extends Reflux.Store {
+  teardown(): void;
+  unsubscribeListeners: Reflux.Subscription[];
+}
+
+export function makeSafeRefluxStore<T extends Reflux.Store>(
+  store: T
+): SafeRefluxStore & T {
+  const originalListenTo = store.listenTo.bind(store);
+  const safeStore = store as SafeRefluxStore & T;
+
+  // Warning: We are going to be overriding instance methods here!
+  // The reason we cannot create a new store and extend it is that
+  // reflux throws whenever createStore() is called with a storeDefinition
+  // that implements listenTo and we cannot create a clone is because
+  // Reflux.createStore implicitly connects the store already and we cannot
+  // disconnect that store and connect our own...
+  safeStore.unsubscribeListeners = [];
+
+  function listenTo(
+    this: SafeRefluxStore,
+    action: Reflux.Listenable,
+    callback: (...data: any) => void
+  ) {
+    const unsubscribeListener = originalListenTo(action, callback);
+    this.unsubscribeListeners.push(unsubscribeListener);
+    return unsubscribeListener;
+  }
+  function teardown(this: SafeRefluxStore) {
+    while (this.unsubscribeListeners.length > 0) {
+      const unsubscribeListener = this.unsubscribeListeners.pop();
+
+      if (unsubscribeListener !== undefined && 'stop' in unsubscribeListener) {
+        unsubscribeListener.stop();
+        return;
+      }
+
+      throw new Error(
+        `Attempting to call ${JSON.stringify(
+          unsubscribeListener
+        )}. Unsubscribe listeners should only include function calls`
+      );
```

Comment:
```suggestion
      const listenerName = JSON.stringify(unsubscribeListener);
      throw new Error(
        `Attempting to call ${listenerName}. Unsubscribe listeners should only include function calls`
      );
```

nit: maybe a tad easier to read like this? Not sure if prettier will wrap it again anyway if it's too long

Reply:
👍🏼 
```tsx
      const stringifiedListenerType = JSON.stringify(unsubscribeListener);
      throw new Error(
        `Attempting to call ${stringifiedListenerType}. Unsubscribe listeners should only include function calls`
      );
```